avm/res/desktop-virtualization/application-group/modules/application.bicep

```diff
@@ -0,0 +1,87 @@
+metadata name = 'Azure Virtual Desktop Application Group Application'
+metadata description = 'This module deploys an Azure Virtual Desktop Application Group Application.'
+metadata owner = 'Azure/module-maintainers'
+
+@sys.description('Conditional. The name of the parent Application Group to create the application(s) in. Required if the template is used in a standalone deployment.')
+param applicationGroupName string
+
+@sys.description('Required. Name of the Application to be created in the Application Group.')
+param name string
+
+@sys.description('Optional. Description of Application..')
+param description string = ''
+
+@sys.description('Required. Friendly name of Application..')
+param friendlyName string
+
+@sys.description('Required. Specifies a path for the executable file for the application.')
+param filePath string
+
+@allowed([
+  'Allow'
+  'DoNotAllow'
+  'Require'
+])
+@sys.description('Optional. Specifies whether this published application can be launched with command-line arguments provided by the client, command-line arguments specified at publish time, or no command-line arguments at all.')
+param commandLineSetting string = 'DoNotAllow'
+
+@sys.description('Optional. Command-Line Arguments for Application.')
+param commandLineArguments string = ''
+
+@sys.description('Optional. Specifies whether to show the RemoteApp program in the RD Web Access server.')
+param showInPortal bool = false
+
+@sys.description('Optional. Path to icon.')
+param iconPath string = ''
+
+@sys.description('Optional. Index of the icon.')
+param iconIndex int = 0
+
+@sys.description('Optional. Enable telemetry.')
+param enableTelemetry bool = true
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.desktopvirtualization-appgroup.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource appGroup 'Microsoft.DesktopVirtualization/applicationGroups@2023-09-05' existing = {
+  name: applicationGroupName
+}
+
+resource application 'Microsoft.DesktopVirtualization/applicationGroups/applications@2023-09-05' = {
```

Comment:
This should be created as a child-module as opposed to a 'nested' template. The main reason is that we may opt to publish child modules in the future + want to keep it consistent with other modules.

To this end, please move this file into a folder `avm/res/desktop-virtualization/application-group/application/main.bicep` that would also contain a ReadMe.md that the `Set-AVMModule` utility can generate for you.

Also note, that the reference in the parent module would need to be updated to the changed folder path.